pkg/config/types/downloaders.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+import (
+	"slices"
+	"strings"
+)
+
+var _ Provider = (*ResultDownloaders)(nil)
+
+type ResultDownloaders struct {
+	Disabled []string       `yaml:"Disabled,omitempty"`
+	Timeout  Duration       `yaml:"Timeout,omitempty"`
+	IPFS     IpfsDownloader `yaml:"IPFS,omitempty"`
+}
+
+func (r ResultDownloaders) Enabled(kind string) bool {
+	return !slices.ContainsFunc(r.Disabled, func(s string) bool {
+		return strings.ToLower(s) == strings.ToLower(kind)
+	})
+}
```

Comment:
Here's how it ends up being used in practice:
```go
	if cfg.Enabled(models.StorageSourceIPFS) {
		if cfg.IPFS.Installed() {
			ipfsClient, err := ipfs_client.NewClient(ctx, cfg.IPFS.Endpoint)
			if err != nil {
				return nil, err
			}
			providers[models.StorageSourceIPFS] = ipfs.NewIPFSDownloader(ipfsClient)
		}
	}
```

`Enabled` checks if the feature is allowed to operate, and `Installed` ensures the feature is actually present and operational. In this setup, IPFS could be enabled but not installed, which is why the distinction makes sense.

It feels like we might be overcomplicating things a bit here, but I’m happy to adjust if you think it’s necessary.
Since `IsDisabled` is your preferred choice, I’ll update it accordingly to match your expectations.